docs/output_file_model_documentation.md

```diff
@@ -0,0 +1,818 @@
+# OutputFile Model Documentation
+
+The OutputFile model in Metamist represents and tracks file outputs from bioinformatics analyses. This system automatically validates files against Google Cloud Storage (GCS), extracts metadata, and manages complex file relationships like VCF files with their index files.
+
+---
+
+## Table of Contents
+
+- [Quick Start](#quick-start)
+  - [Upload a Single File](#upload-a-single-file)
+  - [Upload a VCF with Index](#upload-a-vcf-with-index)
+  - [Understanding the Response](#understanding-the-response)
+- [Understanding the Model](#understanding-the-model)
+  - [What is an OutputFile?](#what-is-an-outputfile)
+  - [Data Flow Overview](#data-flow-overview)
+  - [File Validation](#file-validation)
+  - [Parent-Child Relationships](#parent-child-relationships)
+- [API Usage Guide](#api-usage-guide)
+  - [REST API](#rest-api)
+  - [GraphQL API](#graphql-api)
+  - [Output Structures Comparison](#output-structures-comparison)
+  - [Response Format Examples](#response-format-examples)
+- [Advanced Patterns](#advanced-patterns)
+  - [Complex Nested Structures](#complex-nested-structures)
+  - [Handling Invalid Files](#handling-invalid-files)
+  - [Secondary Files](#secondary-files)
+  - [Special File Types](#special-file-types)
+- [Database Schema Reference](#database-schema-reference)
+  - [Entity Relationship Diagram](#entity-relationship-diagram)
+  - [Table Structures](#table-structures)
+  - [Model Fields](#model-fields)
+
+---
+
+## Quick Start
+
+This section gets you started with the most common use cases for OutputFiles in Metamist.
+
+### Upload a Single File
+
+The simplest way to add an output file to an analysis is to include it in the `outputs` field when creating or updating an analysis.
+
+**REST API Example:**
+
+```bash
+POST /analysis/{project}/
```

Comment:
Should we be showing the graphql examples by default to encourage use of that rather than the REST api?